app/controllers/concerns/order_stock_check.rb

```diff
@@ -24,23 +24,41 @@ def check_order_cycle_expiry(should_empty_order: true)
     return unless current_order_cycle&.closed?
 
     Alert.raise_with_record("Notice: order cycle closed during checkout completion", current_order)
-    if should_empty_order
-      current_order.empty!
-      current_order.assign_order_cycle! nil
-    end
 
-    flash[:info] = I18n.t('order_cycle_closed')
+    handle_closed_order_cycle(should_empty_order)
```

Comment:
I think the code would read better like this :
```suggestion
    handle_closed_order_cycle if should_empty_order
```